submissions/evgeniy24/friends_app/app.js

```diff
@@ -0,0 +1,234 @@
+    friendCity.setAttribute('target', '_blank');
+
+    moreWrap.append(friendPhone, friendEmail);
+    mainInfo.append(friendsName, friendAge, friendCity);
+    friendPhotoWrap.append(friendPhoto);
+    cardWrap.append(friendPhotoWrap, mainInfo, moreWrap);
+    FRIENDS_LIST_WRAP.append(cardWrap);
+}
+
+//find a friend by the name ------------------------------------
+SEARCH_FIELD.addEventListener('input', function() {
+    let searchName = SEARCH_FIELD.value.toLowerCase();
+    if (searchName !== '') {
+            searchResultList = friendsList.filter(function(friend) {
+            return friend.name.first.toLowerCase().includes(searchName) 
+        })
+        return renderFriendsList(searchResultList);
+    }  else {
+        renderFriendsList(friendsList);
+    }
+})
+
+//sorting List of friends --------------------------------------------
+SORTING_MENU.addEventListener('click', function(event) {
+    if (searchResultList) {
+        sortedList = searchResultList;
+    } else if (filteredByGenderList) {
+        sortedList = filteredByGenderList;
+    } else {
+        sortedList = [...friendsList];
+    }
+
+    const sortByName = function(a, b) {
+        const nameA = a.name.first.toUpperCase();
+        const nameB = b.name.first.toUpperCase();
+        return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+    }
+
+    const sortByAge = function(a, b) {
+        return (a.dob.age - b.dob.age);
+    }
+    
+    if (event.target === SORT_BY_NAME_AZ) {
+        sortedList.sort((a, b) => sortByName(a, b));
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_NAME_ZA) {
+        sortedList.sort((a, b) => sortByName(b, a));
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_AGE_YO) {
+        sortedList.sort((a, b) => sortByAge(a, b));
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_AGE_OY) {
+        sortedList.sort((a, b) => sortByAge(b, a));
+        renderFriendsList(sortedList);
+    } 
+})
+
+//filtering list of friends by gender
+FILTER_MENU.addEventListener('click', function(event) {
+    if (searchResultList) {
+        genderNeutralList = searchResultList;
+    } else {
+        genderNeutralList = [...friendsList];
+    }
+
+    if (event.target === FILTER_BY_MALE) {
+        filteredByGenderList = genderNeutralList.filter(function(friend) {
+           return friend.gender === 'male'; 
+        })
+        renderFriendsList(filteredByGenderList);
+        
+    } else if (event.target === FILTER_BY_FEMALE) {
+        filteredByGenderList = genderNeutralList.filter(function(friend) {
+           return friend.gender === 'female'; 
+        })
+        renderFriendsList(filteredByGenderList);
+
+    } else if (event.target === FILTER_BY_ALL) {
+        filteredByGenderList = [...friendsList];
+        renderFriendsList(filteredByGenderList);
+        }
```

Comment:
Redundant spacing. Use prettier :)

Reply:
I thought I was using. 
it was installed but not activated ))